added backend RG

winmike opened the pull request.

cenkms commented:
Looks pretty good thanks @winmike !

cykreng commented:
Parametrize location, and other parameters that can pull from workload name, etc.
Changing to App Service Plan Premium to leverage private endpoints

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+    perSiteScaling: false
+    maximumElasticWorkerCount: 1
+    isSpot: false
+    freeOfferExpirationTime: '7/18/2021 9:36:43 PM'
```
The free offer is over, not sure if we'd like to include this

winmike replied:
agree: commented out so far

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+    enabled: true
+    hostNameSslStates: [
+      {
+        name: 'funcappapimcsbackendmicroservicea.azurewebsites.net'
```
this needs to be a parameter

winmike replied:
agree: added as parameter

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+    hostNamesDisabled: false
+    containerSize: 1536
+    dailyMemoryTimeQuota: 0
+    httpsOnly: false
```
I'd strongly suggest we invest some time in getting SSL implemented here. For first iteration I'm fine but we cannot have none secure transport over http imho

winmike replied:
agree: changed to true

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+    hyperV: false
+    siteConfig: {
+      numberOfWorkers: 1
+      linuxFxVersion: 'DOCKER|mcr.microsoft.com/azure-functions/dotnet:3.0-appservice-quickstart'
```
What do we want with DOCKER here?

winmike replied:
it is the descriptor of the tech stack: here the Azure Function App running from docker container (no code to deploy)

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+  // location: 'West Europe'
+  properties: {
+    numberOfWorkers: 1
+    defaultDocuments: [
```
I expect we can drop all these since we're supposed to point to an API

CarloKuip commented on deployment/bicep/backend.bicep:
```diff
@@ -0,0 +1,433 @@
+      'default.aspx'
+      'index.php'
+    ]
+    netFrameworkVersion: 'v4.0'
```
can be removed

winmike replied:
agree: commented out so far

whsalazar commented:
@winmike It looks like you have to review Carlo's requested changes and resolve conflicts in order to merge to main.

winmike commented:
- extracted most important properties into parameters
- changed pre-defined AppSvcPlan SKU to PremiumV2 P2v2
- applied changes proposed by CarloKuip
- checked with az cli on 20210811
